util/fibers/simple_channel.h

```diff
@@ -126,6 +130,68 @@ template <typename T, typename Q> void SimpleChannel<T, Q>::StartClosing() {
   pop_ec_.notifyAll();
 }
 
+// SimpleQueue-like interface, but also keeps track over the size of Ts it owns.
+// It has a slightly less efficient TryPush() API as it forces construction of Ts even if they are
+// not pushed.
+// T must have a .size() method, which should return the heap-allocated size of T, excluding
+// anything included in sizeof(T). We could generalize this in the future.
+template <typename T, typename Queue = folly::ProducerConsumerQueue<T>> class SizeTrackingChannel {
```

Comment:
I prefer introducing this in the dragonfly codebase.

Reply:
Ok, will do.
Somewhat ironically, the only change I'll need to make is indeed the capacity() method then :)